### PR TITLE
Bugfix WV events

### PR DESCRIPTION
**Pull request by a-cat-named-snowball**
Changes:
 - Fixed bill id formatting
 - Fixed date formatting that was causing scraper to exit with an error

### CODE REVIEW

File: scrapers/wv/events.py [hunk 1]
```diff
@@ -101,6 +97,12 @@ def scrape_meeting_page(self, url):
         when = when.split("-")[0]
         when = self.clean_date(when)
         when.replace("Thursady", "Thursday")
+
+        # Remove all text after the third comma to make sure no extra text
+        # is included in the date. Required to correctly parse texxt like this:
+        # "Friday, March 3, 2023, Following wrap up of morning agenda"
+        when = ",".join(when.split(",")[:2])
```
Comment (showerst) [inline scrapers/wv/events.py:104]: Good idea; can you move this down to the clean_date function where the rest of the date logic lives?

File: scrapers/wv/events.py [hunk 2]
```diff
@@ -132,10 +134,28 @@ def scrape_meeting_page(self, url):
...
+                for bill in bills:
                     bill_id = re.sub(r"\.\s*", "", bill[0], flags=re.IGNORECASE)
                     bill_id = re.sub(r"house bill", "HB", bill_id, flags=re.IGNORECASE)
                     bill_id = re.sub(r"senate bill", "SB", bill_id, flags=re.IGNORECASE)
+
+                    # Final step to set correct number of spaces in the id
+                    components = re.search(
+                        r"([A-Z]+)\s*(\d+)", bill_id, flags=re.IGNORECASE
+                    )
+                    bill_id = f"{components.group(1)} {components.group(2)}"
```
Comment (chrisyamas) [inline scrapers/wv/events.py:157]: Need to adjust the regex to exclude any leading zeros from the numerical component of the `bill_id`.
- i.e. "SB 009" should be "SB 9", "HR 0243" should be "HR 243"

**Review by chrisyamas — CHANGES_REQUESTED** [review]
Once you've made the requested changes, this should be good to merge.

File: scrapers/wv/events.py (hunk 2 above)
Comment (a-cat-named-snowball) [reply scrapers/wv/events.py]: Fixed with https://github.com/openstates/openstates-scrapers/pull/4521/commits/4ffa4ce024bf4158035c68261f91b97a089860cb

File: scrapers/wv/events.py (hunk 1 above)
Comment (a-cat-named-snowball) [reply scrapers/wv/events.py]: Fixed with https://github.com/openstates/openstates-scrapers/pull/4521/commits/34d9c100bd25867cc4956bcfcbe46dc9eefcf7fa

File: scrapers/wv/events.py [hunk 3]
```diff
@@ -132,17 +128,40 @@ def scrape_meeting_page(self, url):
...
+                    flags=re.IGNORECASE,
+                )
+
+                for bill in bills:
                     bill_id = re.sub(r"\.\s*", "", bill[0], flags=re.IGNORECASE)
                     bill_id = re.sub(r"house bill", "HB", bill_id, flags=re.IGNORECASE)
                     bill_id = re.sub(r"senate bill", "SB", bill_id, flags=re.IGNORECASE)
+
+                    # Final step to set correct number of spaces in the id
+                    components = re.search(
```
Comment (johnseekins) [inline scrapers/wv/events.py:148]: It's possibly a premature optimization, but typically we want to pre-compile regexes if we're going to use them in loops:

```
component_re = re.compile(r"([A-Z]+)\s*(\d+)")
for bill in bills:
    components = component_re.search(bill_id, re.IGNORECASE)
```

And so on.

Comment (johnseekins) [reply scrapers/wv/events.py:148]: If it would be a pain to fix all these regexes, don't let it stop a merge. But something to consider generally.

Comment (a-cat-named-snowball) [reply scrapers/wv/events.py]: Fixed with https://github.com/openstates/openstates-scrapers/pull/4521/commits/d4f0ae7ef37b73579fccc2cb962f36619d7688da

Comment (chrisyamas) [reply scrapers/wv/events.py]: @a-cat-named-snowball for the sake of consistency, can you do the same for those three patterns on lines 145-147, that is, pre-compile them before the for loop?

File: scrapers/wv/events.py [hunk 4]
```diff
@@ -132,17 +128,43 @@ def scrape_meeting_page(self, url):
...
+
+                component_re = re.compile(r"([A-Z]+)\s*(\d+)", flags=re.IGNORECASE)
+                period_and_whitespace_re = re.compile(r"\.\s*", flags=re.IGNORECASE)
+                house_bill_re = re.compile(r"house bill", flags=re.IGNORECASE)
+                senate_bill_re = re.compile(r"senate bill", flags=re.IGNORECASE)
+
+                for bill in bills:
+                    bill_id = re.sub(period_and_whitespace_re, "", bill[0])
+                    bill_id = re.sub(house_bill_re, "HB", bill_id)
+                    bill_id = re.sub(senate_bill_re, "SB", bill_id)
```
Comment (chrisyamas) [inline scrapers/wv/events.py:150]: These three just need to be reformatted in the form of: `bill_id = period_and_whitespace_re.sub("", bill[0])` etc

Comment (a-cat-named-snowball) [reply scrapers/wv/events.py]: Oops, sorry didn't catch that, fixed now.

**Review by chrisyamas — APPROVED** [review]
Just ran it again and data looks good. Thanks for those regex fixes! Merging now.

**Pull request merged**